src/optim.jl

```diff
@@ -19,6 +19,14 @@ function optimize!(@nospecialize(job::CompilerJob), mod::LLVM.Module; opt_level=
         end
         run!(pb, mod, tm)
     end
+
+    # Make sure any lingering TLS getters are rewritten even if upstream LLVM passes
+    # transformed them before the GPULowerPTLSPass had a chance to run.
+    if occursin("StaticCompilerTarget", string(typeof(job.config.target))) &&
```

Comment:
This is the opposite of the code pattern we use, we should never hard code a specific target here, but rather expose hooks for a target to configure the pipeline